yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -418,11 +418,16 @@ object YarnSparkHadoopUtil {
    *
    * @return The correct OOM Error handler JVM option, platform dependent.
    */
-  def getOutOfMemoryErrorArgument: String = {
+  def getOutOfMemoryErrorArgument(sparkConf: SparkConf, javaOpts: ListBuffer[String]): String = {
     if (Utils.isWindows) {
       escapeForShell("-XX:OnOutOfMemoryError=taskkill /F /PID %%%%p")
     } else {
-      "-XX:OnOutOfMemoryError='kill %p'"
+      val onOOME = javaOpts.find(x => x.contains("-XX:OnOutOfMemoryError"))
+      if (onOOME == None) {
```

Comment:
There are several things wrong with this Scala code, but more generally, this isn't a good way to design this method. If you really mean to optionally add an argument, then see how things like the PermGen argument are handled and make it work more that way.